### PR TITLE
Set the output variable as sensitive

### LINKED ISSUE #1: Consider marking this output value as sensitive
Regarding https://github.com/andrewsav-datacom/vault-octopus/blob/e67d45bd42370c41a01b9fc1876a459b60215ada/vault-get-secret.ScriptBody.ps1#L91 - consider marking this as a sensitive output variable. The effect of this is that the variable will be masked in any logs, and propagated through the Octopus deployment pipeline as a sensitive value. This lessens the risk of sensitive values being exposed.

https://octopus.com/docs/deployment-process/variables/output-variables#sensitive-output-variables

We shipped this in https://github.com/OctopusDeploy/Issues/milestone/370 as part of https://github.com/OctopusDeploy/Issues/issues/4541

### PR DESCRIPTION
**Pull request by michaelnoonan**
Fixes #1

### CODE REVIEW

File: vault-get-secret.ScriptBody.ps1 [hunk 1]
```diff
@@ -88,5 +93,9 @@ $res | Foreach-Object {
     $name = $prefix + $label + $_.name
 
     "Writing ##{Octopus.Action[#{Octopus.Step.Name}].Output.$name}" | Write-Host
-    Set-OctopusVariable -name $name -value $_.value
+    if (sensitiveOutputVariablesSupported -eq $true) {
```
Comment (matt-richardson) [inline vault-get-secret.ScriptBody.ps1]:
```suggestion
    if ($sensitiveOutputVariablesSupported) {
```

File: vault-get-secret.ScriptBody.ps1 [hunk 2]
```diff
@@ -72,6 +72,11 @@ if ("#{token}" -ne "##{token}") {
     Invoke-WebRequest -Headers @{"X-Vault-Token"="$vault_token"} "#{vaultUrl}/v1/auth/token/revoke-self" -Method post -UseBasicParsing | Out-Null
 }
 
+$sensitiveOutputVariablesSupported = ((Get-Command 'Set-OctopusVariable').Parameters.GetEnumerator() | Where-Object { $_.key -eq "Sensitive" }) -ne $null
+if ($sensitiveOutputVariablesSupported -ne $true) }
```
Comment (matt-richardson) [inline vault-get-secret.ScriptBody.ps1]:
```suggestion
if (-not $sensitiveOutputVariablesSupported) }
```

**Comment by michaelnoonan** [comment]
Thanks to @matt-richardson for the hint on feature detection! :)

**Comment by michaelnoonan** [comment]
Ping @andrewsav-datacom  - do you like the look of this change? Happy to help out where I can. :)

**Comment by andrewsav-bt** [comment]
@matt-richardson @michaelnoonan somehow I have not seen it until now. I'll make sure to have a look soon.

**Comment by andrewsav-bt** [comment]
Guys, @matt-richardson @michaelnoonan , just a quick note that I did not forget. Usually I can turn around this type of stuff pretty quickly but in this case there were _circumstances_. It's still on my to do list and I will action it soon.

**Comment by andrewsav-bt** [comment]
Got the error below when tried to update existing projects to use the new script. Note that this error is not script specific, but I cannot progress with testing it until this is resolved. I contacted Octopus Deploy support with this issue. Thanks.
```
Octopus v2018.9.0

Sequence contains more than one matching element
System.InvalidOperationException
   at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at Octopus.Server.Web.Api.Actions.DeploymentActionUpdate.ActionTemplateActionsUpdateResponder.<>c__DisplayClass5_2.<ExecuteRegistered>b__4(DeploymentAction a)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Octopus.Server.Web.Api.Actions.DeploymentActionUpdate.ActionTemplateActionsUpdateResponder.ExecuteRegistered(String actionTemplateId)
   at Octopus.Server.Web.Infrastructure.Api.Responder`1.Respond(TDescriptor options, NancyContext context)
   at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at CallSite.Target(Closure , CallSite , Object , Object , NancyContext )
   at Octopus.Server.Web.Infrastructure.OctopusNancyModule.<>c__DisplayClass14_0.<get_Routes>b__1(Object x)
   at CallSite.Target(Closure , CallSite , Func`2 , Object )
   at Nancy.Routing.Route.<>c__DisplayClass4.<Wrap>b__3(Object parameters, CancellationToken context)
```

**Comment by andrewsav-bt** [comment]
@michaelnoonan @matt-richardson 

```
At C:\Octopus\Work\20181127064424-50147-3\Script.ps1:83 char:44 
+ if (-not $sensitiveOutputVariablesSupported) } 
+                                            ~ 
Missing statement block after if ( condition ). 
At C:\Octopus\Work\20181127064424-50147-3\Script.ps1:83 char:46 
+ if (-not $sensitiveOutputVariablesSupported) } 
+                                              ~ 
Unexpected token '}' in expression or statement. 
At C:\Octopus\Work\20181127064424-50147-3\Script.ps1:85 char:1 
+ } 
+ ~ 
Unexpected token '}' in expression or statement. 
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordEx  
   ception 
    + FullyQualifiedErrorId : MissingStatementBlock 
 
November 27th 2018 19:44:28Fatal
The remote script failed with exit code 1 
November 27th 2018 19:44:28Fatal
The action Get Credentials on the Octopus Server failed 
```

Guys, I'm wondering, did you actually attempt to run this one? It looks like it has a syntax error, could you please have a look? I also would appreciate a lot, if you could test if it actually works, not just update the curly bracket. That is install it to your instance of Octopus Deploy, point to your Vault installation and see if it does what it's supposed to do with regards to sensitiveness variables.

Thank you in advance!

**Comment by andrewsav-bt** [comment]
@michaelnoonan @matt-richardson Hello, guys, I've spend some more time on this pull request and checked it some change, would you like to have a look?

**Comment by michaelnoonan** [comment]
Oh cool! Sorry @andrewsav-datacom this fell off my radar until you pinged. You caught me - I didn't end up testing it. A truly low quality drive by from me. 😊 

This looks like it's in good shape. Hope it helps!

**Pull request merged**